src/main/java/com/tstepnik/planner/service/TaskService.java

```diff
@@ -0,0 +1,119 @@
+package com.tstepnik.planner.service;
+
+import com.tstepnik.planner.domain.task.Importance;
+import com.tstepnik.planner.domain.task.Task;
+import com.tstepnik.planner.domain.user.User;
+import com.tstepnik.planner.repository.TaskRepository;
+import com.tstepnik.planner.repository.UserRepository;
+import org.springframework.stereotype.Service;
+
+import javax.persistence.EntityNotFoundException;
+import java.security.Principal;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class TaskService {
+
+    private final Importance DEFAULT_IMPORTANCE = Importance.LITTLE_IMPORTANT;
+
+    private final TaskRepository taskRepository;
+    private final UserRepository userRepository;
+
+    public TaskService(TaskRepository taskRepository, UserRepository userRepository) {
+        this.taskRepository = taskRepository;
+        this.userRepository = userRepository;
+    }
+
+    public List<Task> getTasks() {
+        archiveTasks();
+        return taskRepository.findAll();
+    }
+
+    public List<Task> getAllLoggedUserTasks(Principal principal) {
+        User loggedUser = userRepository.findByLogin(principal.getName()).orElseThrow(()
+                -> new EntityNotFoundException("User not found"));
+        archiveTasks();
+        return taskRepository.findAllByUser(loggedUser);
+    }
+
+    public List<Task> getAllLoggedUserTasksOrderByImportance(Principal principal) {
+        User loggedUser = userRepository.findByLogin(principal.getName()).orElseThrow(()
+                -> new EntityNotFoundException("User not found"));
+        archiveTasks();
+        return taskRepository.findAllByUserOrderByImportance(loggedUser);
+    }
+
+    public List<Task> getLoggedUserParticularTasks(Principal principal, Importance importance) {
+        User loggedUser = userRepository.findByLogin(principal.getName()).orElseThrow(()
+                -> new EntityNotFoundException("User not found"));
+        archiveTasks();
+        return taskRepository.findAllByUserIdaAndImportance(loggedUser.getId(), importance);
+    }
+
+    public List<Task> getUserTasks(Long userId) {
+        archiveTasks();
+        return taskRepository.findALLByUserId(userId);
+    }
+
+    public Task getTaskById(Long taskId) {
+        archiveTasks();
```

Comment:
This method should check if the user is admin if so, return task, if not check if logged user is owner and return if he is, if not throw access denied exception.